src/transformers/models/llama/tokenization_llama.py

```diff
@@ -433,3 +418,26 @@ def _build_conversation_input_ids(self, conversation: "Conversation") -> List[in
             f"{B_INST} {(dialogue[-1][1]).strip()} {E_INST}", add_special_tokens=False
         )
         return dialog_tokens
+
+    @property
+    def default_chat_template(self):
```

Comment:
This Jinja magic is a bit hard to parse for me. WDYT about having a docstring that shows an example of the template (maybe it could also just live in a guide)?

Reply:
Yeah, this is a good point! I might factor some of it back out into `tokenizer.template_args`. That would make the whole thing more readable, and mean that editing the underlying template could be saved for more advanced use cases. 

Also, a lot of the untidiness you're seeing in the default template here is because I have to maintain backwards compatibility with the existing template and options - templates on future models could be a lot simpler!

Reply:
After further thinking, I prefer to leave `template_args` out and just pack everything into the template. I wrote a doc [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_25323/en/chat_templating) though, which I think makes it much more intuitive!

Reply:
I think the docstring of each `default_chat_template` should show expected formatting or at least explain what is happening! ( + it's tested)

Reply:
Doubling down, could you put back the helpful:
```
        ```
        <bos>[INST] B_SYS SytemPrompt E_SYS Prompt [/INST] Answer <eos>
        <bos>[INST] Prompt [/INST] Answer <eos>
        <bos>[INST] Prompt [/INST]
        ```
```
think a readable string can always help!